templates/fine-tune-llm_v2/cookbooks/lora_vs_full_parameter/README.md

```diff
@@ -0,0 +1,98 @@
+# Fine-Tuning with LoRA vs full-parameter
+
+**⏱️ Time to complete**: 45 minutes (for the LoRA fine-tuning)
+
+This guide assumes that you have familiarized yourself with the [main fine-tuning guide](../../README.md) of this template.
+In this cookbook, we explain the nuances of fine-tuning with [LoRA (Low-Rank Adaptation)](https://arxiv.org/abs/2106.09685) versus full-parameter fine-tuning.
+
+## Quick theoretical comparison
+
+Full-parameter fine-tuning takes the LLM "as is" and trains it on the given dataset. In principle, this is regular supervised training like in the pretraining stage of the LLM. You can expect full-parameter fine-tuning to result in slightly higher model quality.
+
+[LoRA (Low-Rank Adaptation)](https://arxiv.org/abs/2106.09685) is a fine-tuning technique that freezes all the weights of your LLM and adds a few parameters to it that get fine-tuned instead. These additional parameters make up a LoRA checkpoint. There are two important things to take away from this:
+1. Since all the original weights are frozen, they don't have to be optimized and therefore don't take up as many resources during fine-tuning. In practice, you can fine-tune on a smaller cluster.
+2. Since the checkpoint only consists of the few additional parameters, it is very small. If we load the original model into memory, we can swap out the fine-tuned weights quickly. Therefore, it makes for an efficient scheme for serving many fine-tuned models alongside each other.
+
+You can find a more in-depth analysis of this topic [here](https://www.anyscale.com/blog/fine-tuning-llms-lora-or-full-parameter-an-in-depth-analysis-with-llama-2).
+
+## How to configure LoRA vs full-parameter fine-tuning jobs
+
+Both fine-tuning techniques require the same dataset format and result in a checkpoint that you can serve with Anyscale's serving template.
+Next, we illustrate this by showing you two commands that fine-tune on the same data - once with LoRA, once with full-parameter fine-tuning.
+You can look at the respective yaml files to see how they differ in their configuration.
+
+### How to launch a LoRA fine-tuning job
+
+
+```python
+# [Optional] You can set the WandB API key to track model performance
+# !export WANDB_API_KEY={YOUR_WANDB_API_KEY}
+
+# Fine-tune Llama 3 8B with LoRA
+!python main.py llama-3-8b-lora.yaml
```

Comment:
we should already have lora vs. full param in the config list one level above this. So why not reuse them?

Reply:
This won't work with the deepspeed config, because that's a relative path as well.
We could manipulate that in main but I think it's wise here to keep it simple since all of this is code that users can end up reading and get confused about.

Reply:
I made it work here. https://github.com/anyscale/templates/blob/711038265881b1188ea6ccfbacaad94bef35a973/templates/fine-tune-llm_v2/cookbooks/continue_from_checkpoint/llama-3-8b.yaml#L17

Let's adopt this pattern instead of doing `../../deepspeed_configs`. You need to make it clear that the user should run the command from the root of the template dir.